src/parser.yy

```diff
@@ -236,6 +236,7 @@ expr : INT             { $$ = new ast::Integer($1); }
      | expr PTR ident  { $$ = new ast::FieldAccess(new ast::Unop(token::MUL, $1), $3); }
      | "(" IDENT ")" expr %prec CAST  { $$ = new ast::Cast($2, false, $4); }
      | "(" IDENT MUL ")" expr %prec CAST  { $$ = new ast::Cast($2, true, $5); }
+     | expr"["expr"]" { $$ = new ast::ArrayAccess($1, $3); }
```

Comment:
Nitpick: could you add the spacing between elements like the other rules have. I feel it makes it easier to count them to match to `$1` and `$3`

Reply:
Added this as requested.